src/woperator.jl

```diff
@@ -139,13 +139,31 @@ function Base.:\(W::WOperator, x::Number)
     end
 end
 
-function LinearAlgebra.mul!(Y::AbstractVecOrMat, W::WOperator, B::AbstractVecOrMat)
+function LinearAlgebra.mul!(Y::AbstractVector, W::WOperator, B::AbstractVector)
     # Compute mass_matrix * B
     if isa(W.mass_matrix, UniformScaling)
         a = -W.mass_matrix.λ / W.gamma
         @. Y=a*B
     else
-        mul!(_vec(Y), W.mass_matrix, vec(B))
+        mul!(Y, W.mass_matrix, B)
```

Comment:
are you sure this actually works without this? `Y` as a matrix wouldn't work anymore, so what happens for example with `u0` a matrix?